test/services/katello/registration_manager_test.rb

```diff
@@ -275,29 +275,15 @@ def test_registration_existing_host
         ::Katello::RegistrationManager.register_host(@host, rhsm_params, [@content_view_environment])
       end
 
-      def test_unregister_host_without_katello_agent
+      def test_unregister_host
         @host = FactoryBot.create(:host, :with_content, :with_subscription, :content_view => @content_view,
                                    :lifecycle_environment => @library, :organization => @content_view.organization)
-        ::Katello.expects(:with_katello_agent?).returns(false)
         @host.content_facet.expects(:cves_changed?).returns(false)
         ::Katello::Resources::Candlepin::Consumer.expects(:destroy)
         ::Katello::EventQueue.expects(:push_event).never
         ::Katello::RegistrationManager.unregister_host(@host, :unregistering => true)
       end
 
-      def test_unregister_host_with_katello_agent
```

Comment:
I was admiring your work when it occurred to me that there's always a small chance that after all this cleanup is done, there may be "delete host agent queue" events in the db which will break the event queue when they get processed because their handler class will be removed. You'll  need an upgrade rake task to delete those from the events table

Reply:
Yeah, I was just thinking about delete_host_agent_queue and how to handle that..  :cry:

Reply:
Do we need that? Isn't the queue emptied when upgrading any way?

Reply:
The queue isn't explicitly emptied on upgrade and there could be events on it especially for this specific case because it's scheduled to run ten minutes in the future. Any hosts unregistered ~10 minutes before shutdown will put the events on queue and if Katello is upgraded with that event removed it'll raise an error around an unsupported event. There could also be events stuck on the queue due to this bug: #10692 

Looking at the code: https://github.com/Katello/katello/blob/c9fe25d6705d5cf84dc60fefb521e8d3e939688b/app/services/katello/event_queue.rb#L73-L75

An error gets raised if an unsupported event is found on queue which is what will happen when the event's class is removed from the app. However, that error may be rescued, logged, and the event removed: https://github.com/Katello/katello/blob/c9fe25d6705d5cf84dc60fefb521e8d3e939688b/app/services/katello/event_monitor/poller_thread.rb#L61-L70

Definitely test that to make sure it doesn't break the queue though.

Reply:
So, do you think I should add to the rescue here instead of the upgrade rake task?

Either way, can this be addressed in a followup issue?

Reply:
I think the rescue that's there will handle the situation already - but that needs a confirmation - definitely outside of the scope here!